common/sysinfo_linux.go

```diff
@@ -0,0 +1,154 @@
+			re := regexp.MustCompile(` +`)
+			value := strings.Split(string(re.ReplaceAll([]byte(scanner.Text()), []byte(" "))), " ")[1]
+			multiplerStr := strings.Split(string(re.ReplaceAll([]byte(scanner.Text()), []byte(" "))), " ")[2]
+
+			if multiplerStr == "kB" {
+				multipler = 1024
+			} else {
+				// "/proc/meminfo" output always in kB only. If we are getting different string, something wrong.
+				err := fmt.Errorf("Meminfo value is not in kB")
+				return 0, err
+			}
+
+			if result, err = strconv.Atoi(value); err != nil {
+				return 0, err
+			}
+			return int64(result) * int64(multipler), nil
+		}
+	}
+
+	if err = cmd.Wait(); err != nil {
+		err := fmt.Errorf("GetMemInfo failed with error: %v", err)
+		return 0, err
+	}
+
+	return 0, nil
+}
+
+// parseStat parse the output of stat and return common structure for stat and statx
+func parseStat(in unix.Stat_t) UnixStatContainer {
+	var out UnixStatContainer
+
+	// Fileds not present in stat call
+	out.statx = false
+	out.mask = 0
+	out.attributes = 0
+	out.attributesMask = 0
+	out.birthTime = time.Time{}
+
+	// Fields present in stat call
+	out.numLinks = in.Nlink
+	out.groupGID = in.Gid
+	out.ownerUID = in.Uid
+	out.accessTime = time.Unix(in.Atim.Sec, in.Atim.Nsec)
+	out.changeTime = time.Unix(in.Ctim.Sec, in.Ctim.Nsec)
+	out.devID = in.Dev
+	out.iNode = in.Ino
+	out.modTime = time.Unix(in.Mtim.Sec, in.Mtim.Nsec)
+	out.mode = in.Mode
+	out.size = uint64(in.Size)
+	out.repDevID = in.Rdev
+	return out
+}
+
+// parseStatx parse the output of statx and return common structure for stat and statx
+func parseStatx(in unix.Statx_t) UnixStatContainer {
+	var out UnixStatContainer
+
+	out.statx = true
+	out.mask = in.Mask
+	out.attributes = in.Attributes
+	out.attributesMask = in.Attributes_mask
+	out.birthTime = time.Unix(in.Btime.Sec, int64(in.Btime.Nsec))
+
+	// Fields present in stat call
+	out.numLinks = uint64(in.Nlink)
+	out.groupGID = in.Gid
+	out.ownerUID = in.Uid
+	out.accessTime = time.Unix(in.Atime.Sec, int64(in.Atime.Nsec))
+	out.changeTime = time.Unix(in.Ctime.Sec, int64(in.Ctime.Nsec))
+	out.devID = unix.Mkdev(in.Dev_major, in.Dev_minor)
+	out.iNode = in.Ino
+	out.modTime = time.Unix(in.Mtime.Sec, int64(in.Mtime.Nsec))
+	out.mode = uint32(in.Mode)
+	out.size = in.Size
+	out.repDevID = unix.Mkdev(in.Rdev_major, in.Rdev_minor)
+
+	return out
+}
+
+// GetExtendedProperties return stat/statx properties of file/folder.
+func GetExtendedProperties(fileName string) (UnixStatAdapter, error) {
+	{ // attempt to call statx, if ENOSYS is returned, statx is unavailable
+		var stat unix.Statx_t
+		// dirfd is a null pointer, because we should only ever be passing relative paths here, and directories will be passed via transferInfo.Source.
+		// AT_SYMLINK_NOFOLLOW is not used, because we automagically resolve symlinks. TODO: Add option to not follow symlinks, and use AT_SYMLINK_NOFOLLOW when resolving is disabled.
+		err := unix.Statx(0, fileName,
+			unix.AT_STATX_SYNC_AS_STAT,
```

Comment:
I think we should use AT_SYMLINK_NOFOLLOW since we are a low level function and caller if he wants can resolve symlink before calling GetExtendedProperties().
let's discuss this.